src/actions/Action.tsx

```diff
@@ -50,21 +67,39 @@ function Action({
     return <ErrorAlert errorMessage={`Action doesn't exist`} />;
   }
 
-  const runActionBtn = (
-    <Button
-      type="primary"
-      onClick={() => runAction()}
-      disabled={!canBeRun}
-      loading={isRunActionLoading}
-      icon={hasBeenRun ? <CheckCircleOutlined /> : null}
-    >
-      Run Action
-    </Button>
+  const extraButtons = (
+    <Space size="middle">
+      <Button
+        type="primary"
+        onClick={() => runAction()}
+        disabled={!canBeRun}
+        loading={isRunActionLoading}
+        icon={hasBeenRun ? <CheckCircleOutlined /> : <PlaySquareOutlined />}
+      >
+        Run Action
+      </Button>
+      <Popconfirm
+        title="Are you sure to delete this Action?"
+        onConfirm={() => deleteAction()}
+        okText="Yes"
+        cancelText="No"
+        placement="left"
+      >
+        <Button
```

Comment:
**nit:** When actions are running you can make this button disabled as we don't allow deletion in this case, so it could be misleading that someone delete it, but the action is still running.